extensions/jdbc/jdbc-postgresql/deployment/src/main/java/io/quarkus/jdbc/postgresql/deployment/PostgreSQLJDBCReflections.java

```diff
@@ -20,6 +20,29 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         final String driverName = "org.postgresql.Driver";
         reflectiveClass.produce(ReflectiveClassBuildItem.builder(driverName).build());
 
+        // We want to register these postgresql "object" types since they are used by a driver to build result set elements
+        // and reflection is used to create their instances. While ORM might only use a `PGInterval` if a @JdbcType(PostgreSQLIntervalSecondJdbcType.class)
+        // is applied to a Duration property, we still register other types as users might create their own JdbcTypes that
+        // would rely on some subtype of a PGobject:
+        final String[] pgObjectClasses = new String[] {
+                "org.postgresql.util.PGobject",
+                "org.postgresql.util.PGInterval",
+                "org.postgresql.util.PGmoney",
+                "org.postgresql.geometric.PGbox",
+                "org.postgresql.geometric.PGcircle",
+                "org.postgresql.geometric.PGline",
+                "org.postgresql.geometric.PGlseg",
+                "org.postgresql.geometric.PGpath",
+                "org.postgresql.geometric.PGpoint",
+                "org.postgresql.geometric.PGpolygon",
+                // one more subtype of the PGobject, it doesn't look like that this one will be instantiated through reflection,
+                // but let's have it here anyway:
+                "org.postgresql.jdbc.PgResultSet.NullObject"
```

Comment:
If it's not necessary, I don't see why we would do this.

Registering types for reflection has a cost.